### ISSUE TITLE
[BUG] PNJ Recycleur

### ISSUE THREAD

**Issue by Drykqo** [RivrsMc/SeasonSky-Issues#471]
### Confirmation

- [X] J'ai bien vérifié si le bug a déjà été report
- [X] Le bug que je report n'est pas un bug de duplication (discord ticket pour cela)

### Pseudo en jeu

Litchs

### Pseudo discord

triopsL's#5883

### Que s'est-il passé ?

un pnj qui a une interaction précise mais qui ne fait pas l'interaction lorsqu'on réunit ce qui est demandé, rien ne se passe, on ne perd rien mais rien ne nous est donné

### Étapes à suivre pour reproduire le bug

Faire /warps prendre Garage, aller le voir le PNJ "Karl" qui se trouve 20 blocs a coté, réunir (exemple) 10 boulbi jaune et 1 engrenage pour faire l'échange avec le pnj, --> Bug l'interaction/ échange ne se fait pas

### Qu'aurait-il dû arriver ?

le pnj aurait dû procédé a l'échange comme prévu

### Date et heure de la production du bug

09-10 18:00:44

### Serveur

Spawn-02

### Information supplémentaire

UUID: e37ff4eb-878e-4184-bfa0-fdb46df46217
Position: 181 141 -1183 world
Ping: 27ms

**Comment by LupusArctos4** [comment]
Si c'est une quete alors c'est normal elles sont buger et ils sont au courant

**Comment by Drykqo** [comment]
Ce n’est pas une quête, c’est un pnj du spawn qui est bug pour certains échanges, certains fonctionnent d’autres non

**Comment by LupusArctos4** [comment]
C'est quoi les autres échanges qu'il peut faire ?
Tu les a tester depuis la mise à jour ?

**Comment by Drykqo** [comment]
Oui j’ai retester, les échanges contre des bonbons à l’orange fonctionne, mais j’en ai tester 2 pour les bonbons à la fraise et ça n’a pas fonctionner donc j’imagine que tous les échanges pour les bonbons à la fraise sont bug, si tu as besoin d’un screen des échanges qu’il propose demande moi

**Comment by LupusArctos4** [comment]
Je veux bien le screen parce que c'est peut être un problème de nbt tags
Questions tu l'avais tous fait avant la mise à jour ?

**Comment by Drykqo** [comment]
comment ça je l'avais tous fait avant la MAJ ?
![2023-10-10_13 56 24](https://github.com/RivrsMc/SeasonSky-Issues/assets/147425719/d2700542-9d1f-4909-8c09-404b5471b592)
![2023-10-10_13 56 26](https://github.com/RivrsMc/SeasonSky-Issues/assets/147425719/3cfb691a-24fe-4746-a6aa-83fb02131e0f)
![2023-10-10_13 56 33](https://github.com/RivrsMc/SeasonSky-Issues/assets/147425719/a0b08fee-d07e-46f6-b0b5-bb4602140b8b)
![2023-10-10_13 56 34](https://github.com/RivrsMc/SeasonSky-Issues/assets/147425719/ae7cf800-1956-4d4e-9935-0fa893937c54)

**Comment by LupusArctos4** [comment]
Ok merci, ce que je veux dire c'est : est-ce que tu avais tester avant le 1er octobre ou c'est récemment que tu ne peux plus le faire ?

**Comment by Drykqo** [comment]
Non, je n'avais pas essayer avant, ce n'est que récemment que j'ai voulu utiliser ce pnj

**Comment by LupusArctos4** [comment]
Ok donc oui je pense que c'est un pb de nbt tag
D'ailleurs tu peux modif le nom de ton issue au moins les dev pourront savoir plus facilement ce que c'est, je pense que tu peux mettre : [BUG] PNJ recycleur

**Comment by Drykqo** [comment]
C'est fait, je vous laisse gérer la suite alors, si tu as d'autres questions n'hésite pas

**Comment by LupusArctos4** [comment]
Je ne suis pas dans le staff personnellement mais j'ai beaucoup beaucoup l'habitude de faire des issues donc vu que là c'était un peu flou si je peux aider les développeurs à le corriger vite c'est toujours mieux je trouve

**Comment by Drykqo** [comment]
ahhhh je vois

**Comment by LupusArctos4** [comment]
@Drykqo on vient de me dire que pour que ça fonctionne il faut que les pets aient les même multiplicateur, en d'autres termes il faut qu'ils soit en stack de 10 (quand ils ont le même multiplicateur tu peux les stacks)

**Comment by LupusArctos4** [comment]
Mais là pour corriger le problème il faudrait que le pnj accepte l'entité du nom du pet et non le pet avec ses caractéristiques intrinsèques ce qui je pense est compliqué parce que avec des items renames c'est peut être possible de contourner le système

**Comment by Drykqo** [comment]
Eh bien ils n’ont qu’à créer des « fragments » de bonbon, avec un pnj pour craft un bonbon avec 10 fragments

**Comment by Drykqo** [comment]
Par ce que là, 10 pets c’est déjà compliqué alors si en plus il faut le même multiplicateur c’est juste pas possible

**Comment by LupusArctos4** [comment]
L'idée de créer des fragments n'est pas bête peut résoudre ce problème effectivement

**Comment by LupusArctos4** [comment]
L'idée de créer des fragments n'est pas bête peut résoudre ce problème effectivement

**Comment by Asherate** [comment]
@ArDiDEx je viens de regarder la config, il y a uniquement les materials et custom model data, donc je comprend pas pourquoi le multiplicateur du pet y jouerais un rôle, tu aurais une petite idée ?

**Comment by LupusArctos4** [comment]
@Asherate la description de l'item ne rentre pas en compte justement ?

**Comment by Asherate** [comment]
@LupusArctos4 non, uniquement le type d'item, par exemple PAPER, et son custommodeldata (qui lui donne sa texture custom)

**Comment by LupusArctos4** [comment]
@Asherate ah oui d'accord bas oui pour le coup je ne comprends pas pourquoi ça fait ça
j'ai peut etre une idée mais à voir je ne suis pas sûr, est-ce parce qu'il faut qu'il soit stack par 10 pour que ca fonctionne ?
si c'est pour ca, l'idée de mettre un item custom pour 1 item, par exemple du sucre et apres avec le sucre vous faites un craft d'item :
[ 0; 2; 0;
2; 2; 2;
0; 2; 0]
ca revient a 10 comme avant mais vous réglé le problème

**Comment by ArDiDEx** [comment]
> @ArDiDEx je viens de regarder la config, il y a uniquement les materials et custom model data, donc je comprend pas pourquoi le multiplicateur du pet y jouerais un rôle, tu aurais une petite idée ?

Hello! C'est car le plugin check que l'item soit stack par le nombre requis, hormis en faisant une logique encore plus compliqué que celle actuelle j'ai pas moyen de changer ça facilement
